src/main/java/polymorphism/FunctionalStaticPolymorphism.java

```diff
@@ -83,8 +94,13 @@ public static Object wordsAmount(Object a, Object b, Object c, Object d) {
      * @param predicates predicates to filter words
      * @return map of words and their amount
      */
-    public static Object wordsAmount(Object a, Object... predicates) {
-        throw new ExerciseNotCompletedException();
+    public static Map<String, Integer> wordsAmount(String text, Predicate<? super String>... predicates) {
+        return Arrays.stream(text.split(" "))
+                .filter(not(String::isBlank))
+                .map(String::toLowerCase)
+                .filter(s -> predicates == null || Arrays.stream(predicates).allMatch(p -> p.test(s)))
```

Comment:
Predicate has method `add` pls take a look at that and it will give you time complexity O(n) instead of O(n*m)